### PR TITLE
Couchbase v2.6.3 & Opentelemetry v1.16.0

### PR DESCRIPTION
**Pull request by dynastymasra**
1. Opentelemetry v1.16.0 doesn't have metrictest and is not backward compatibility. 

2. Handle uint32 for couchbase retry request.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,12 +1,25 @@
...
+	github.com/go-logr/logr v1.2.4 // indirect
+	github.com/go-logr/stdr v1.2.2 // indirect
+	github.com/golang/snappy v0.0.4 // indirect
+	github.com/google/uuid v1.3.0 // indirect
+	github.com/pmezard/go-difflib v1.0.0 // indirect
+	golang.org/x/sys v0.8.0 // indirect
+	gopkg.in/yaml.v3 v3.0.1 // indirect
+)
+
+go 1.20
```
Comment (chvck) [inline go.mod:25]: Does this have to be bumped to 1.20 or can it be left at 1.16?

**Comment by chvck** [comment]
Hi @dynastymasra this is great, thank you. 

Have you signed our CLA? Instructions can be found at https://github.com/couchbase/couchbase-jvm-clients/blob/master/CONTRIBUTING.md#pull-requests - unfortunately this does require signing up to our gerrit platform. I'll try to check again today to ensure that you do actually need to do this step.

File: go.mod (hunk 1 above)
Comment (dynastymasra) [reply go.mod:25]: It requires go1.18 or later, Because opentelemtry is using generic types.

**Comment by dynastymasra** [comment]
Hi @chvck  ,

Yes, I have signed the CLA.
Thank you in advance

**Comment by chvck** [comment]
Great, thanks for your contribution!

**Review by chvck — APPROVED** [review]

**Pull request merged**